arangod/IResearch/IResearchView.cpp

```diff
@@ -1393,11 +1404,20 @@ arangodb::Result IResearchView::commit() {
     memoryStore._writer->clear(); // prepare the store for reuse
 
     SCOPED_LOCK(_toFlush->_readMutex); // do not allow concurrent read since _storePersisted/_toFlush need to be updated atomically
+    const auto readerBeforeReopen = _storePersisted._reader;
```

Comment:
same here: does this copy or reference?

Reply:
reader is just a pimpl, so the copy is cheap and noexcept

Reply:
:+1: